exporter/signalfxexporter/dpclient.go

```diff
@@ -48,37 +51,81 @@ type sfxDPClient struct {
 
 func (s *sfxDPClient) pushMetricsData(
 	ctx context.Context,
-	md consumerdata.MetricsData,
+	md pdata.Metrics,
 ) (droppedTimeSeries int, err error) {
-	accessToken := s.retrieveAccessToken(md)
+	metricsData := pdatautil.MetricsToMetricsData(md)
+	var numDroppedTimeseries int
+	var errs []error
+
+	var currentToken string
+	var batchStartIdx int
```

Comment:
I wonder if the previous map-based approach isn't worth using and optimizing in the future if necessary.  Considering this approach doesn't improve the (near) worse cases by much and io will likely always be the bottleneck here it seems premature to abandon it.  The map would ensure a single request per token where there could be hundreds in the worse case here

Reply:
We could improve it further later. But this approach at lease doesn't degrade performance for any of the use cases, especially for those when the keys are now being passed. 
With the previous approach, if user doesn't have batching enabled, we'd have the same number of pushes + additional map and slice allocations per push.

Reply:
I agree the current approach is not optimized for the worst case. This approach kind of expects that on average the worst case scenario is not a common occurrence. If in the future that assumption ends up being invalid we should update. I'd like to get this patch in as a first step since it's an improvement from what we currently have.